src/System.Private.CoreLib/shared/System/AppContext.cs

```diff
@@ -8,12 +8,15 @@
 using System.Runtime.Loader;
 using System.Runtime.Versioning;
 using System.Threading;
+using System.IO;
+using System.Reflection;
+using System.Diagnostics;
 
 namespace System
 {
     public static partial class AppContext
     {
-        private static readonly Dictionary<string, object?> s_dataStore = new Dictionary<string, object?>();
+        private static Dictionary<string, object?> s_dataStore;
```

Comment:
What is the benefit of this change?  There are common access paths to AppContext that don't end up accessing this?

Reply:
Is the benefit that we get to presize the dictionary appropriately because we know the count in Setup?

Reply:
@stephentoub I was moving those two methods here and noticed that I can just make the code a bit smaller by removing the static constructor.
> Is the benefit that we get to presize the dictionary appropriately because we know the count in Setup?

As far as I understand, Dictionary will use `3` as a default capacity if it's not specified.
so if setup passes more than 3 - it will have to do re-alloc(s).